staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go

```diff
@@ -115,6 +135,14 @@ type CustomResourceDefinitionStatus struct {
 	// AcceptedNames are the names that are actually being used to serve discovery
 	// They may be different than the names in spec.
 	AcceptedNames CustomResourceDefinitionNames
+
+	// StoredVersions are all versions ever marked as storage in spec. Tracking these
```

Comment:
actually, more like "are all versions that were ever written to the storage (etcd)".

Reply:
well it is something in between actually. It is all versions that ever got a persist request (storage request may still fail but we store the version here anyway). But your description is closer to reality :)